### DISCUSSION TITLE
[Feature] Resizing and rotating polygons in GUI

### DISCUSSION THREAD

**Discussion by smihael** [wkentaro/labelme#1081] [Ideas / Feature requests]
**Is your feature request related to a problem? Please describe.**
Labelme provides a very easy way to move around the polygons by simply dragging them (left mouse click, hold) in the GUI. This proves to be very useful when tagging many similar images as once can keep the annotations from previous image when editing next image. However, very often the polygons have also to be rotated or scaled for accurate annotation.

**Describe the solution you'd like**
A bounding box could be drawn around the mask with an edge that enables resizing and rotating the polygon.

A similar solution can be seen in many office programs:

![resizing](https://user-images.githubusercontent.com/2089122/129020021-176e562a-b960-42ac-acb1-5e82b72d5713.jpg)
![rotating](https://user-images.githubusercontent.com/2089122/129020036-0c831683-d2ba-4fa1-8c5e-23c6eea075fd.jpg)

**Comment by ryouchinsa** [comment]
RectLabel is an offline image annotation tool for object detection and segmentation.
Although this is not an open source program, with RectLabel you can rotated the polygon using the Rotate mode, and scale up/down the polygon from the center dragging on the polygon pressing the option key.
If necessary, you can export/import the Labelme JSON format.
https://rectlabel.com/help#rotate_box

![draw_obb](https://user-images.githubusercontent.com/1954306/214553302-80e3976f-cf6f-48cd-a9fd-282828a35bb8.jpg)
